predicators/envs/base_env.py

```diff
@@ -229,31 +229,42 @@ def _load_task_from_json(self, json_file: Path) -> EnvironmentTask:
         """
         with open(json_file, "r", encoding="utf-8") as f:
             json_dict = json.load(f)
-        # Parse objects.
-        type_name_to_type = {t.name: t for t in self.types}
         object_name_to_object: Dict[str, Object] = {}
-        for obj_name, type_name in json_dict["objects"].items():
-            obj_type = type_name_to_type[type_name]
-            obj = Object(obj_name, obj_type)
-            object_name_to_object[obj_name] = obj
-        assert set(object_name_to_object).issubset(set(json_dict["init"])), \
-            "The init state can only include objects in `objects`."
-        assert set(object_name_to_object).issuperset(set(json_dict["init"])), \
-            "The init state must include every object in `objects`."
-        # Parse initial state.
-        init_dict: Dict[Object, Dict[str, float]] = {}
-        for obj_name, obj_dict in json_dict["init"].items():
-            obj = object_name_to_object[obj_name]
-            init_dict[obj] = obj_dict.copy()
-        init_state = utils.create_state_from_dict(init_dict)
+        if not CFG.override_json_with_input:
```

Comment:
can we modify this to say `override_json_goal_with_input`? You're not modifying the initial state at all right? So probably good to make that clear in the flag